source/xprod.js

```diff
@@ -24,11 +24,11 @@ var xprod = _curry2(function xprod(a, b) { // = xprodWith(prepend); (takes about
   var ilen = a.length;
   var j;
   var jlen = b.length;
-  var result = [];
+  var result = Array(ilen + jlen);
```

Comment:
This should be multiplication instead of addition yes? `Array(ilen * jlen)`. I suspect the huge performance gains for xprod have something to do with this..

Reply:
Updated